src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java

```diff
@@ -56,15 +55,15 @@ else if (flag != Flag.INTERNAL) {
             if ((flag == Flag.RESET && isCountDirectionDown()) || (Long)value > getMaximumTime() + 500 && (!isCountDirectionDown() || flag != Flag.RECALCULATE)) {
                 return getMaximumTime();
             }
-            if ((flag == Flag.RESET && !isCountDirectionDown()) || (Long)value < getMinimumTime() - 500) {
-                return getMinimumTime();
+            if ((flag == Flag.RESET && !isCountDirectionDown()) || (Long)value < 0 - 500) {
```

Comment:
Why not `-500` directly?

Reply:
I did it this way so it matches the pattern of the if above it.